etl-pipeline/managers/db.py

```diff
@@ -44,94 +44,90 @@ def generateEngine(self):
         except Exception as e:
             raise e
 
-    def initializeDatabase(self):
+    def initialize_database(self):
         if not inspect(self.engine).has_table('works'):
             Base.metadata.create_all(self.engine)
 
-    def createSession(self, autoflush=False):
+    def create_session(self, autoflush=False):
         if not self.engine:
-            self.generateEngine()
+            self.generate_engine()
         self.session = sessionmaker(bind=self.engine, autoflush=autoflush)()
 
-    def startSession(self):
+    def start_session(self):
         self.session.begin_nested()
 
-    def commitChanges(self, retry=False):
+    def commit_changes(self, retry=False):
         try:
             self.session.commit()
         except OperationalError as oprErr:
             logger.error('Deadlock in database layer, retry batch')
             logger.debug(oprErr)
 
-            self.rollbackChanges()
+            self.rollback_changes()
             
             if retry is False:
-                self.commitChanges(retry=True)
+                self.commit_changes(retry=True)
             else:
                 logger.warning('Already retried batch, dropping')
 
-    def rollbackChanges(self):
+    def rollback_changes(self):
         self.session.rollback()
 
-    def closeConnection(self):
-        self.commitChanges()
-        self.session.close()
-        self.engine.dispose()
-
     def close_connection(self):
+        self.commit_changes()
```

Comment:
hmm - what are your thoughts on this side effect? I think every client of the db_manager should explicitly call commit_changes before closing the connection.